sunpy/map/map_factory.py

```diff
@@ -128,7 +100,11 @@ def _read_file(self, fname, **kwargs):
         # NOTE: use os.fspath so that fname can be either a str or pathlib.Path
         # This can be removed once read_file supports pathlib.Path
         log.debug(f'Reading {fname}')
-        pairs = read_file(os.fspath(fname), **kwargs)
+        try:
+            pairs = read_file(os.fspath(fname), **kwargs)
+        except Exception as e:
+            warnings.warn(f'failed to read {fname}',SunpyUserWarning)
+            raise
```

Comment:
I'm not sure a warn and a raise makes any sense. 

The error from the failed attempt at reading a file should already be sent to the user.

Reply:
I think what we want to do to is more selective filtering based on the exception raised.

If the file does not exist, raise and stop working but if the file fails to be read then raise a warning but continue.

Reply:
> I'm not sure a warn and a raise makes any sense.
> 
> The error from the failed attempt at reading a file should already be sent to the user.

issue was "if one file fails then it kills the whole load without telling you which file it failed on."

Initially @jmason86 had a TypeError and error doesn't tell which file caused the error.My attempt was to print name of the file that failed to load.

Reply:
> I think what we want to do to is more selective filtering based on the exception raised.
> 
> If the file does not exist, raise and stop working but if the file fails to be read then raise a warning but continue.

By default , If file doesn't exist it raise and stop working.

If the reading of one file fails, Now it raise warning and stop working. So I think instead of raising error for `read_file( )` it's better to skip there.

Reply:
But if all we want is to report what file failed, we can just add text to the raised expectation and that would be enough.

Reply:
It's better to skip files that failed to read as you suggested.
then all it need is to remove the `raise Exception` in current solution.

`read_file` is the part of `sunpy.io.file_tools` that reads the file.

So is this the better approach.